lib/viewmodels/account_details_model.dart

```diff
@@ -6,61 +6,77 @@ import 'package:app/locator.dart';
 import 'package:app/services/auth.dart';
 import 'package:app/services/google_location_search_service.dart';
 import 'package:app/services/organisation_service.dart';
+import 'package:app/services/analytics.dart';
 
 class AccountDetailsViewModel extends BaseModel {
-  
-  final AuthenticationService _authenticationService = locator<AuthenticationService>();
-  final GoogleLocationSearchService _googleLocationSearchService = locator<GoogleLocationSearchService>();
-  final OrganisationService _organisationService = locator<OrganisationService>();
-  
+  final AuthenticationService _authenticationService =
+      locator<AuthenticationService>();
+  final GoogleLocationSearchService _googleLocationSearchService =
+      locator<GoogleLocationSearchService>();
+  final OrganisationService _organisationService =
+      locator<OrganisationService>();
+  final Analytics _analyticsService = locator<Analytics>();
+
   String _name;
   set name(String name) => _name = name;
-  DateTime _dob; 
-  set dob(DateTime dob) { 
+  DateTime _dob;
+  set dob(DateTime dob) {
     _dob = dob;
     dobFieldController.text = dateToString(dob);
   }
+
   String _location;
-  set location(String location) { 
+  set location(String location) {
     _location = location;
     locationFieldController.text = location;
   }
+
   String _orgCode;
   set orgCode(String orgCode) => _orgCode = orgCode;
-  
+
   String get placeSearchUuid => _googleLocationSearchService.sessionToken;
   // The dob in the form or if null the current known dob
   DateTime get latestDob => _dob ?? currentUser.getDateOfBirth();
 
-  Organisation get userOrganisation => _authenticationService.currentUser.organisation;
+  Organisation get userOrganisation =>
+      _authenticationService.currentUser.organisation;
 
   final _formKey = GlobalKey<FormState>();
   GlobalKey get formKey => _formKey;
   final _dobFieldController = TextEditingController();
   TextEditingController get dobFieldController => _dobFieldController;
   final _locationFieldController = TextEditingController();
   TextEditingController get locationFieldController => _locationFieldController;
-  
+
   void init() {
     _dobFieldController.text = dateToString(currentUser.getDateOfBirth());
     _locationFieldController.text = currentUser.getLocation();
   }
-  
+
   void save() async {
     setBusy(true);
     await _authenticationService.updateUserDetails(
-        name: _name, dob: _dob, orgCode: _orgCode
-    ); 
+        name: _name, dob: _dob, orgCode: _orgCode);
+    setBusy(false);
+    notifyListeners();
+  }
+
+  void delete() async {
+    setBusy(true);
+    await _authenticationService.deleteUserAccount();
```

Comment:
Can we add some error handling in here?

So for example 
```suggestion
    if (!await _authenticationService.deleteUserAccount()) {
        _dialog_service -> something when wrong message
    } else {
        await _analyticsService.logUserAccountDeleted();
        logout();
    }
```

Reply:
Yes definitely agree. And as you say we can explore the topic after Christmas.